tests/functional/object/mcg/ui/test_bucket_policy_ui.py

```diff
@@ -0,0 +1,205 @@
+from ocs_ci.ocs.ui.page_objects.bucket_tab_permissions import (
+    BucketsTabPermissions,
+    PolicyConfig,
+    PolicyType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@black_squad
+@post_upgrade
+class TestBucketPolicyUI:
+    """
+    Test class for bucket policy UI operations
+    """
+
+    @pytest.mark.parametrize(
+        "policy_config",
+        [
+            pytest.param(
+                {
+                    "name": "AllowPublicReadAccess",
+                    "method": "set_bucket_policy_ui",
+                    "params": {},
+                },
+                marks=[tier1, polarion_id("OCS-7382")],
+            ),
+            pytest.param(
+                {
+                    "name": "AllowAccessToSpecificAccount",
+                    "method": "set_bucket_policy_specific_account_ui",
+                    "params": {"account_list": ["123456789012"]},
+                },
+                marks=[tier3, polarion_id("OCS-7383")],
+            ),
+            pytest.param(
+                {
+                    "name": "EnforceSecureTransportHTTPS",
+                    "method": "set_bucket_policy_enforce_https_ui",
+                    "params": {},
+                },
+                marks=[tier3, polarion_id("OCS-7384")],
+            ),
+            pytest.param(
+                {
+                    "name": "AllowReadWriteAccessToFolder",
+                    "method": "set_bucket_policy_folder_access_ui",
+                    "params": {
+                        "folder_path": "documents",
+                        "account_list": ["123456789012"],
+                    },
+                },
+                marks=[tier3, polarion_id("OCS-7385")],
+            ),
+        ],
+    )
+    def test_set_bucket_policy_ui(self, setup_ui_class_factory, policy_config):
+        """
+        Test setting various bucket policies via UI.
+
+        Args:
+            policy_config (dict): Configuration containing policy name, method, and parameters
+
+        Raises:
+            pytest.skip: If no buckets are available for testing
+        """
+        setup_ui_class_factory()
+        policy_name = policy_config["name"]
+        method_name = policy_config["method"]
+        params = policy_config["params"]
+
+        logger.info(f"Starting test to set {policy_name} bucket policy")
+
+        bucket_ui = BucketsTab()
+        bucket_ui.navigate_buckets_page()
+
+        account_dependent_policies = [
+            "AllowAccessToSpecificAccount",
+            "AllowReadWriteAccessToFolder",
+        ]
+
+        target_bucket_name = None
+
+        if policy_name in account_dependent_policies:
+            initial_buckets = set(bucket_ui.get_buckets_list())
+
+            bucket_ui.create_bucket_ui(method="obc")
+
+            bucket_ui.navigate_buckets_page()
+            bucket_ui.page_has_loaded(sleep_time=2)
+
+            for attempt in range(6):
```

Comment:
@AviadP could you please use variable instead of hardcoded "6"? For example num_of_attempts ?